runtime/interpreter/decode.go

```diff
@@ -20,141 +20,55 @@ package interpreter
 
 import (
 	"fmt"
-	"io"
 	"math"
-	"math/bits"
-	"strconv"
-	"strings"
 
 	"github.com/fxamacker/cbor/v2"
-
+	"github.com/onflow/atree"
 	"github.com/onflow/cadence/runtime/common"
-	"github.com/onflow/cadence/runtime/common/orderedmap"
 	"github.com/onflow/cadence/runtime/sema"
 )
 
-type DecodingCallback func(value interface{}, path []string)
-
-// A DecoderV5 decodes CBOR-encoded representations of values.
-// It can decode storage format version 4 and later.
-//
-type DecoderV5 struct {
-	decoder        *cbor.StreamDecoder
-	owner          *common.Address
-	version        uint16
-	decodeCallback DecodingCallback
-	isByteDecoder  bool
-}
-
-// maxInt is math.MaxInt32 or math.MaxInt64 depending on arch.
-const maxInt = 1<<(bits.UintSize-1) - 1
+var CBORDecMode = func() cbor.DecMode {
```

Comment:
This is exported, so we can use it e.g. in flow-go in a state migration without having to duplicate it like curently